src/LibraryManager/Providers/FileSystem/FileSystemProvider.cs

```diff
@@ -105,6 +105,14 @@ public ILibraryCatalog GetCatalog()
             {
                 return new LibraryInstallationResult(desiredState, PredefinedErrors.PathOutsideWorkingDirectory());
             }
+            catch (Exception ex) when (ex is ResourceDownloadException || ex.InnerException is ResourceDownloadException)
```

Comment:
`ex.InnerException` could be null

Reply:
if `ex.InnerException` is null here, it will fail the `null is Type` check.

Reply:
Code removed